### PR TITLE
feat: Integration orders with api

### PR DESCRIPTION
**Pull request by Vsevolod1020**

### CODE REVIEW

File: src/app/modules/admin/components/orders/orders.component.ts [hunk 1]
```diff
@@ -23,10 +24,14 @@ export class OrdersComponent implements OnInit, OnDestroy {
...
 
-    constructor(readonly store: Store<AdminState>) {}
+    constructor(
+        readonly store: Store<AdminState>,
+        readonly ordersServerService: OrdersServerService,
+    ) {}
 
     ngOnInit(): void {
         this.store.dispatch(OrdersActions.getOrders());
+        this.ordersServerService.getOrders();
```
Comment (glebek2h) [inline src/app/modules/admin/components/orders/orders.component.ts]: remove it from here

File: src/app/modules/admin/state/effects/orders.effects.ts [hunk 2]
```diff
@@ -11,54 +11,19 @@ export class OrdersEffects {
...
+                    ),
             ),
-            map(orders => OrdersActions.getOrdersSuccess({ orders })),
         );
     });
 
-    constructor(private actions$: Actions) {}
+    constructor(
+        private actions$: Actions,
+        private ordersServerService: OrdersServerService,
```
Comment (glebek2h) [inline src/app/modules/admin/state/effects/orders.effects.ts]: readonly

File: src/app/modules/admin/state/effects/orders.effects.ts [hunk 3]
```diff
@@ -11,54 +11,19 @@ export class OrdersEffects {
...
+                        ),
+                    ),
             ),
-            map(orders => OrdersActions.getOrdersSuccess({ orders })),
         );
     });
 
-    constructor(private actions$: Actions) {}
+    constructor(
+        private actions$: Actions,
```
Comment (glebek2h) [inline src/app/modules/admin/state/effects/orders.effects.ts]: readonly

File: src/app/modules/admin/components/orders/orders.component.ts [hunk 4]
```diff
@@ -23,10 +24,14 @@ export class OrdersComponent implements OnInit, OnDestroy {
         .select(OrdersSelect.selectState)
         .pipe(takeUntil(this.unsubscribe$));
 
-    constructor(readonly store: Store<AdminState>) {}
+    constructor(
+        readonly store: Store<AdminState>,
```
Comment (glebek2h) [inline src/app/modules/admin/components/orders/orders.component.ts]: private

File: src/app/modules/admin/components/orders/orders.component.ts [hunk 5]
```diff
@@ -23,10 +24,14 @@ export class OrdersComponent implements OnInit, OnDestroy {
...
+        readonly ordersServerService: OrdersServerService,
```
Comment (glebek2h) [inline src/app/modules/admin/components/orders/orders.component.ts]: private

File: src/app/modules/admin/state/reducers/orders.reducer.ts [hunk 6]
```diff
@@ -14,7 +14,7 @@ export const reducer = createReducer(
     on(OrdersActions.getOrdersSuccess, (state, action) => {
         return {
             ...state,
-            items: action.orders.items,
+            orders: action.orders.orders,
```
Comment (glebek2h) [inline src/app/modules/admin/state/reducers/orders.reducer.ts]: looks weird, make it just action.orders

File: src/environments/environment.prod.ts [hunk 7]
```diff
@@ -1,3 +1,6 @@
 export const environment = {
     production: true,
+    langKey: 'lang',
+    defaultLang: 'en',
+    api_url: 'http://localhost:3000'
```
Comment (glebek2h) [inline src/environments/environment.prod.ts:5]: we shouldn't hardcode localhost as our route
what will be when we will deploy our app to some domain?

File: src/environments/environment.ts [hunk 8]
```diff
@@ -1,5 +1,6 @@
 export const environment = {
     production: false,
     langKey: 'lang',
-    defaultLang: 'en'
+    defaultLang: 'en',
+    api_url: 'http://localhost:3000'
```
Comment (glebek2h) [inline src/environments/environment.ts:5]: we shouldn't hardcode localhost as our route
what will be when we will deploy our app to some domain?

File: src/environments/environment.prod.ts (hunk 7 above)
Comment (Vsevolod1020) [reply src/environments/environment.prod.ts:5]: I see, but it's just better to change in one place, isn't it? Or maybe there is another way to do it?

File: src/app/modules/admin/components/orders/orders.component.ts [hunk 9]
```diff
@@ -23,7 +24,9 @@ export class OrdersComponent implements OnInit, OnDestroy {
         .select(OrdersSelect.selectState)
         .pipe(takeUntil(this.unsubscribe$));
 
-    constructor(readonly store: Store<AdminState>) {}
+    constructor(
+        private store: Store<AdminState>,
```
Comment (glebek2h) [inline src/app/modules/admin/components/orders/orders.component.ts]: readonly

File: src/app/modules/admin/state/effects/orders.effects.ts [hunk 10]
```diff
@@ -11,54 +11,17 @@ export class OrdersEffects {
...
+                        map(orders => OrdersActions.getOrdersSuccess(orders)),
+                    ),
             ),
-            map(orders => OrdersActions.getOrdersSuccess({ orders })),
         );
     });
 
-    constructor(private actions$: Actions) {}
+    constructor(
+        readonly actions$: Actions,
```
Comment (glebek2h) [inline src/app/modules/admin/state/effects/orders.effects.ts]: private

File: src/app/services/orders-service/orders-server.service.ts [hunk 11]
```diff
@@ -0,0 +1,18 @@
...
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs';
+import { APIResponse, Orders } from 'src/app/modules/admin/state/admin.model';
+import * as serviceConstants from '../service.constants';
+
+@Injectable({
+    providedIn: 'root',
+})
+export class OrdersServerService {
+    constructor(readonly http: HttpClient) {}
```
Comment (glebek2h) [inline src/app/services/orders-service/orders-server.service.ts]: private

**Pull request merged**